### PR TITLE
Handle listeners and certmapping

### PR DESCRIPTION
**Pull request by Annih**
Add LWRP to setup winrm listener
Add `listeners` recipe to configure them by attributes
Add LWRP to setup winrm service certmapping

### CODE REVIEW

**Comment by rveznaver** [comment]
You have no tests.

**Comment by brugidou** [comment]
+1 we can deal with chefspec tests later it's a bit hard to do for now and we aren't even using the cookbook properly yet

File: libraries/base_resource.rb [hunk 1]
```diff
@@ -18,7 +18,11 @@
 # limitations under the License.
 #
 module WinrmConfig
+  # Module to extend Chef Resources with helpers and properties' Hash table
+  # It's used by all `winrm-config` resources.
   module BaseResource
+    MAX_INT16 = 65_535 unless defined? MAX_INT16
+    MAX_INT32 = 2_147_483_647unless defined? MAX_INT32
```
Comment (rveznaver) [inline libraries/base_resource.rb]: Missing space.

Comment (rveznaver) [inline libraries/base_resource.rb]: You're not testing for MAX_INT16 and MAX_INT32

**Comment by rveznaver** [comment]
+1

**Pull request merged**